gnome-initial-setup/pages/site/gis-site-search-entry.c

```diff
@@ -0,0 +1,518 @@
+                                        g_value_get_object (value));
+        break;
+      default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+        break;
+    }
+}
+
+static void
+get_property (GObject *object, guint prop_id,
+              GValue *value, GParamSpec *pspec)
+{
+  GisSiteSearchEntryPrivate *priv = gis_site_search_entry_get_instance_private (
+    GIS_SITE_SEARCH_ENTRY (object));
+
+  switch (prop_id)
+    {
+      case PROP_SITE:
+        g_value_set_boxed (value, priv->site);
+        break;
+      default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+        break;
+    }
+}
+
+static void
+entry_changed (GisSiteSearchEntry *entry)
+{
+  GisSiteSearchEntryPrivate *priv =
+      gis_site_search_entry_get_instance_private (entry);
+  const gchar *text = gtk_entry_get_text (GTK_ENTRY (entry));
+
+  if (!text || text[0] == '\0')
+    set_site_internal (entry, priv->model, NULL, NULL);
+}
+
+static void
+string_append_safe (GString     *string,
+                    const gchar *field)
+{
+  if (field && field[0] != '\0')
+    g_string_append_printf (string, "%s%s", string->len > 0 ? ", " : "", field);
+}
+
+/* NOTE: this assumes the values provided here are already translated into the
+ * local language
+ */
+static gchar*
+get_site_display_name (GisSite *site)
+{
+  const gchar *id = gis_site_get_id (site);
+  const gchar *facility = gis_site_get_facility (site);
+  const gchar *street = gis_site_get_street (site);
+  const gchar *locality = gis_site_get_locality (site);
+  const gchar *region = gis_site_get_region (site);
+  const gchar *country = gis_site_get_country (site);
+
+  GString *string = g_string_new ("");
+
+  string_append_safe (string, id);
+  string_append_safe (string, facility);
+  string_append_safe (string, street);
+  string_append_safe (string, locality);
+  string_append_safe (string, region);
+  string_append_safe (string, country);
+
+  return g_string_free (string, FALSE);
+}
+
+static void
+set_site_internal (GisSiteSearchEntry *entry,
+                   GtkTreeModel       *model,
+                   GtkTreeIter        *iter,
+                   GisSite            *site)
+{
+  GisSiteSearchEntryPrivate *priv =
+      gis_site_search_entry_get_instance_private (entry);
+
+  g_clear_object (&priv->site);
+
+  g_assert (iter == NULL || site == NULL);
+
+  if (iter)
+    {
+      char *name;
+      gtk_tree_model_get (model, iter,
+                          GIS_SITE_SEARCH_ENTRY_COL_DISPLAY_NAME, &name,
+                          GIS_SITE_SEARCH_ENTRY_COL_SITE, &priv->site,
+                          -1);
+      gtk_entry_set_text (GTK_ENTRY (entry), name);
+      g_free (name);
+    }
+  else if (site)
+    {
+      gchar *display_name = get_site_display_name (site);
+      priv->site = g_object_ref (site);
+      gtk_entry_set_text (GTK_ENTRY (entry), display_name);
+
+      g_free (display_name);
+    }
+  else
+    {
+      priv->site = NULL;
```

Comment:
`priv->site` is already `NULL` at this point.

Reply:
Still need to remove this line

Reply:
Sorry, somehow I missed your original comment on this.